src/ImageSharp/Processing/PixelRowOperation.cs

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Numerics;
+
+namespace SixLabors.ImageSharp.Processing
+{
+    /// <summary>
+    /// A <see langword="delegate"/> representing a user defined processing delegate to use to modify image rows.
+    /// </summary>
+    /// <param name="span">The target row of <see cref="Vector4"/> pixels to process.</param>
+    /// <remarks>The <see cref="Vector4.X"/>, <see cref="Vector4.Y"/>, <see cref="Vector4.Z"/>, and <see cref="Vector4.W"/> fields map the RGBA channels respectively.</remarks>
+    public delegate void PixelRowOperation(Span<Vector4> span);
+
+    /// <summary>
+    /// A <see langword="delegate"/> representing a user defined processing delegate to use to modify image rows.
+    /// </summary>
+    /// <typeparam name="T">
+    /// The type of the parameter of the method that this delegate encapsulates.
+    /// This type parameter is contravariant.That is, you can use either the type you specified or any type that is less derived.
+    /// </typeparam>
+    /// <param name="span">The target row of <see cref="Vector4"/> pixels to process.</param>
+    /// <param name="value">The parameter of the method that this delegate encapsulates.</param>
+    /// <remarks>The <see cref="Vector4.X"/>, <see cref="Vector4.Y"/>, <see cref="Vector4.Z"/>, and <see cref="Vector4.W"/> fields map the RGBA channels respectively.</remarks>
+    public delegate void PixelRowOperation<in T>(Span<Vector4> span, T value);
```

Comment:
Just wanted to double-check if these may conflict with the variants needed for the future `ProcessPixelRowsAs` API, but looks like it's not the case.

```C#
public delegate void PixelRowOperation<in TPixel>(Span<TPixel> span) where TPixel : struct, IPixel<TPixel>;

public delegate void PixelRowOperation<in TPixel, TParam>(Span<TPixel> span, TParam value) where TPixel : struct, IPixel<TPixel>;
```

Reply:
Yeah, didn't think it would be an issue. Thanks for reviewing!